# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ruby/sparc.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

This code is buggy, it should use "ta 0x03" on Linux as well
as the BSD's and Solaris.

"ta 0x83" is a hypervisor trap on Niagara that loads a TLB
entry for supervisor code :-)

  - David Miller <davem@davemloft.net>

--- ruby-1.8.6-p36/defines.h.orig	2007-08-17 09:34:13.000000000 +0200
+++ ruby-1.8.6-p36/defines.h	2007-08-17 09:34:40.000000000 +0200
@@ -218,9 +218,7 @@
 #endif
 # if defined(__sparc_v9__) || defined(__sparcv9) || defined(__arch64__)
 	("flushw")
-# elif defined(linux) || defined(__linux__)
-	("ta  0x83")
-# else /* Solaris, OpenBSD, NetBSD, etc. */
+# else /* Solaris, Linux, OpenBSD, NetBSD, etc. */
 	("ta  0x03")
 # endif /* trap always to flush register windows if we are on a Sparc system */
 	;
